docs/source/examples/braket_mirror_circuit.md

```diff
@@ -0,0 +1,257 @@
+
+```{code-cell} ipython3
+def compile_to_rigetti_gateset(circuit: Circuit) -> Circuit:
+    compiled = Circuit()
+
+    for instr in circuit.instructions:
+        if isinstance(instr.operator, gates.Vi):
+            compiled.add_instruction(gates.Instruction(gates.Rx(-np.pi / 2), instr.target))
+        elif isinstance(instr.operator, gates.V):
+            compiled.add_instruction(gates.Instruction(gates.Rx(np.pi / 2), instr.target))
+        elif isinstance(instr.operator, gates.Ry):
+            compiled.add_instruction(gates.Instruction(gates.Rx(-np.pi / 2), instr.target))
+            compiled.add_instruction(gates.Instruction(gates.Rz(instr.operator.angle), instr.target))
+            compiled.add_instruction(gates.Instruction(gates.Rx(np.pi / 2), instr.target))
+        elif isinstance(instr.operator, gates.Y):
+            compiled.add_instruction(gates.Instruction(gates.Rx(-np.pi / 2), instr.target))
+            compiled.add_instruction(gates.Instruction(gates.Rz(np.pi), instr.target))
+            compiled.add_instruction(gates.Instruction(gates.Rx(np.pi / 2), instr.target))
+        elif isinstance(instr.operator, gates.X):
+            compiled.add_instruction(gates.Instruction(gates.Rx(np.pi / 2), instr.target))
+            compiled.add_instruction(gates.Instruction(gates.Rx(np.pi / 2), instr.target))
+        elif isinstance(instr.operator, gates.Z):
+            compiled.add_instruction(gates.Instruction(gates.Rz(np.pi), instr.target))
+        elif isinstance(instr.operator, gates.S):
+            compiled.add_instruction(gates.Instruction(gates.Rz(np.pi / 4), instr.target))
+        elif isinstance(instr.operator, gates.Si):
+            compiled.add_instruction(gates.Instruction(gates.Rz(-np.pi / 4), instr.target))
+        else:
+            compiled.add_instruction(instr)
+    
+    return compiled
+```
+
+## Define the executor
+
+Now that we have a circuit, we define the `execute` function which inputs a circuit and returns an expectation value - here, the frequency of sampling the correct bitstring.
+
+```{code-cell} ipython3
+def execute(
+    circuit: Circuit,
+    shots: int = 1_000, 
+    s3_folder: Tuple[str, str] = ("bucket", "folder/"),
+) -> float:
+    # Add verbatim compiling so that zero-noise extrapolation can be used.
+    if on_aws:
+        circuit = Circuit().add_verbatim_box(compile_to_rigetti_gateset(circuit))
+    
+    # Run the circuit and return the frequency of sampling the correct bitstring.
+    if on_aws:
+        aws_task = aws_device.run(circuit, s3_folder, disable_qubit_rewiring=True, shots=shots)
+    else:
+        aws_task = aws_device.run(circuit, shots=shots)
+    return aws_task.result().measurement_probabilities.get("".join(map(str, correct_bitstring)), 0.0)
```

Comment:
This is suggestion is just a draft suggestion. **Not tested**.
Here we could try to add noise in the classical simulation:

```suggestion
    # Run the circuit and return the frequency of sampling the correct bitstring.
    if on_aws:
        aws_task = aws_device.run(circuit, s3_folder, disable_qubit_rewiring=True, shots=shots)
    else:
        circuit.apply_gate_noise(braket.circuits.Noise.Depolarizing(probability=0.1))
        aws_task = LocalSimulator("braket_dm").run(circuit, shots=shots)
    return aws_task.result().measurement_probabilities.get("".join(map(str, correct_bitstring)), 0.0)
```

Based on this example: https://github.com/aws/amazon-braket-examples/blob/main/examples/braket_features/Simulating_Noise_On_Amazon_Braket.ipynb

Reply:
Thanks! Let me try this.